### PR TITLE
doc: document Start and Await methods and mocks

### PR DESCRIPTION
**Pull request by scotthart**
part of the work for #7658

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14375)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,17 @@ breaking changes in the upcoming 3.x release. This release is scheduled for
 
 ## v2.26.0 - TBD
 
+### BREAKING TESTING CHANGES
```
Comment (dbolduc) [inline CHANGELOG.md:9]: nit: doesn't matter, but line break?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,17 @@ breaking changes in the upcoming 3.x release. This release is scheduled for
...
+If you don't have tests that `EXPECT_CALL` on methods in `MockConnection` that
```
Comment (dbolduc) [inline CHANGELOG.md:10]: FTR, `ON_CALL` would be ambiguous too. Maybe:

```
If you don't mock Long Running Operations (LRO) in your tests, these changes will not affect you.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,17 @@ breaking changes in the upcoming 3.x release. This release is scheduled for
...
+perform Long Running Operations (LRO), these changes will not affect you.
+
+With the addition of new methods to support starting Long Running Operations
+(LRO) synchronously and awaiting their completion separately, the overload set
+for the preexisting LRO methods have been expanded. Uses of `EXPECT_CALL` that
+do not have matchers for the arguments will be ambiguous. To quickly remedy this
+change instances of `EXPECT_CALL(*mock, Method)` to
+`EXPECT_CALL(*mock, Method(_))`.
```
Comment (dbolduc) [inline CHANGELOG.md:18]: optional: we wouldn't write this, but maybe `s/Method(_)/Method(::testing::_)/` for those who are not familiar with the placeholder.

File: generator/internal/mock_connection_generator.cc [hunk 4]
```diff
@@ -103,10 +103,15 @@ class $mock_connection_class_name$ : public $product_namespace$::$connection_cla
                  Not(IsPaginated))),
          MethodPattern(
              {
+                 {// clang-format off
+    "\n  // Due to additional overloads for this method\n"
+    "  // EXPECT_CALL(*mock, $method_name$) is now ambiguous. Use\n"
+    "  // EXPECT_CALL(*mock, $method_name$(_)) instead.\n"},
```
Comment (dbolduc) [inline generator/internal/mock_connection_generator.cc:109]: We should use doxygen style comments `///` and put the code in code quotes.

And optionally, spell out `::testing::_`.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14375)* status: 0 of 527 files reviewed, 3 unresolved discussions (waiting on @dbolduc)

___
*[`generator/internal/mock_connection_generator.cc` line 109 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14375#gh-1656635976:-O0P65nx9pkCJEVsrOo2:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/618569ab8fccf4ffd649e2a93a3a6ed09e234f15/generator/internal/mock_connection_generator.cc#L109)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

We should use doxygen style comments `///` and put the code in code quotes.

And optionally, spell out `::testing::_`.
</blockquote></details>

Done.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**